lib/pure/json.nim

```diff
@@ -1069,12 +1069,13 @@ when defined(nimFixedForwardGeneric):
     dst = jsonNode.copy
 
   proc initFromJson[T: SomeInteger](dst: var T; jsonNode: JsonNode, jsonPath: var string) =
-    when T is uint|uint64:
+    when T is uint|uint64 or (not defined(js) and int.sizeof == 4):
```

Comment:
this logic can be refined in future work, but at least it's good to re-enable a non-broken i386